tower-http/src/compression_utils.rs

```diff
@@ -335,3 +336,37 @@ where
 }
 
 pub(crate) const SENTINEL_ERROR_CODE: i32 = -837459418;
+
+/// Level of compression data should be compressed with.
+#[non_exhaustive]
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum CompressionLevel {
+    /// Fastest quality of compression, usually produces bigger size.
+    Fastest,
+    /// Best quality of compression, usually produces the smallest size.
+    Best,
+    /// Default quality of compression defined by the selected compression algorithm.
+    Default,
+    /// Precise quality based on the underlying compression algorithms'
+    /// qualities. The interpretation of this depends on the algorithm chosen
+    /// and the specific implementation backing it.
+    /// Qualities are implicitly clamped to the algorithm's maximum.
+    Precise(u32),
+}
+
+impl Default for CompressionLevel {
+    fn default() -> Self {
+        CompressionLevel::Default
+    }
+}
+
+impl From<CompressionLevel> for AsyncCompressionLevel {
+    fn from(level: CompressionLevel) -> Self {
+        match level {
+            CompressionLevel::Fastest => AsyncCompressionLevel::Fastest,
+            CompressionLevel::Best => AsyncCompressionLevel::Best,
+            CompressionLevel::Default => AsyncCompressionLevel::Default,
+            CompressionLevel::Precise(quality) => AsyncCompressionLevel::Precise(quality),
+        }
+    }
+}
```

Comment:
This makes `async_compression` a public dependency (part of the public API). I don't think it is public already? Maybe this conversion should be a private function, instead of a publically-usable `impl`.

Reply:
Done, please let me know if this looks better. 👍🏽